pkg/admission/validator_test.go

```diff
@@ -0,0 +1,226 @@
+package admission
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/lithammer/dedent"
+	"github.com/stretchr/testify/assert"
+	admission "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1alpha1 "sigs.k8s.io/service-apis/apis/v1alpha1"
+)
+
+var decoder = codecs.UniversalDeserializer()
+
+func TestServeHTTPBasic(t *testing.T) {
+	assert := assert.New(t)
+	res := httptest.NewRecorder()
+	handler := http.HandlerFunc(ServeHTTP)
+	req, err := http.NewRequest("POST", "", nil)
+	assert.Nil(err)
+	handler.ServeHTTP(res, req)
+	assert.Equal(400, res.Code)
+	assert.Equal("admission review object is missing\n",
+		res.Body.String())
+}
+
+func TestValidationWebhook(t *testing.T) {
+	for _, apiVersion := range []string{
+		"admission.k8s.io/v1beta1",
+		"admission.k8s.io/v1",
+	} {
+		for _, tt := range []struct {
+			name    string
+			reqBody string
+
+			wantRespCode        int
+			wantSuccessResponse admission.AdmissionResponse
+			wantFailureMessage  string
+		}{
+			{
+				name:               "request with present empty body",
+				wantRespCode:       http.StatusBadRequest,
+				wantFailureMessage: "unexpected end of JSON input\n",
+			},
```

Comment:
Maybe add a couple more here:
1. Malformed JSON
2. Valid JSON but of something other than an AdmissionReview

Reply:
Ok I am adding a function to check for AdmissionReview. Please provide some suggestions on how to improve the perf.